pkg/spec/deploymentconfig.go

```diff
@@ -0,0 +1,162 @@
+	if err != nil {
+		return errors.Wrap(err, "unable to fix configMaps")
+	}
+
+	deploymentConfig.Containers, err = fixContainers(deploymentConfig.Containers, deploymentConfig.Name)
+	if err != nil {
+		return errors.Wrap(err, "unable to fix containers")
+	}
+
+	deploymentConfig.InitContainers, err = fixContainers(deploymentConfig.InitContainers, deploymentConfig.Name)
+	if err != nil {
+		return errors.Wrap(err, "unable to fix init-containers")
+	}
+
+	deploymentConfig.Secrets, err = fixSecrets(deploymentConfig.Secrets, deploymentConfig.Name)
+	if err != nil {
+		return errors.Wrap(err, "unable to fix secrets")
+	}
+
+	return nil
+}
+
+func (deploymentConfig *DeploymentConfigSpecMod) Transform() ([]runtime.Object, []string, error) {
+
+	// Create Kubernetes objects (since OpenShift uses Kubernetes underneath, no need to refactor
+	// this portion
+	runtimeObjects, extraResources, err := deploymentConfig.CreateK8sObjects()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "failed to create Kubernetes objects")
+	}
+
+	// Set appropriate GVK BEFORE adding DeploymentConfig controller
+	// as OpenShift controllers are not available in the Kubernetes controller / setGVK check
+	scheme, err := GetScheme()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "unable to get scheme")
+	}
+
+	// Set's the appropriate GVK
+	for _, runtimeObject := range runtimeObjects {
+		if err := SetGVK(runtimeObject, scheme); err != nil {
+			return nil, nil, errors.Wrap(err, "unable to set Group, Version and Kind for generated Kubernetes resources")
+		}
+	}
+
+	// Create the DeploymentConfig controller!
+	deploy, err := deploymentConfig.createOpenShiftController()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "failed to create Kubernetes Deployment controller")
+	}
+
+	// adding controller objects
+	// deploymentConfig will be nil if no deploymentConfig is generated and no error occurs,
+	// so we only need to append this when a legit deploymentConfig resource is returned
+	if deploy != nil {
+		runtimeObjects = append(runtimeObjects, deploy)
+		log.Debugf("deploymentConfig: %s, deploymentConfig: %s\n", deploy.Name, spew.Sprint(deploymentConfig))
+	}
+
+	if len(runtimeObjects) == 0 {
+		return nil, nil, errors.New("No runtime objects created, possibly because not enough input data was passed")
+	}
+
+	return runtimeObjects, extraResources, nil
+}
+
+// Created the OpenShift DeploymentConfig controller
+func (deploymentConfig *DeploymentConfigSpecMod) createOpenShiftController() (*os_deploy_v1.DeploymentConfig, error) {
+
+	// Set replicas to 1 if not set (0)
+	replicas := deploymentConfig.Replicas
+	if replicas == 0 {
+		replicas = 1
+	}
+
+	return &os_deploy_v1.DeploymentConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "DeploymentConfig",
+			APIVersion: "v1",
+		},
+		ObjectMeta: deploymentConfig.ObjectMeta,
+		Spec: os_deploy_v1.DeploymentConfigSpec{
+			Replicas: replicas,
+			Selector: deploymentConfig.Labels,
```

Comment:
shouldn't we use something a bit more specific as a selector?
maybe we could add new label to deployment "deploymentconfig: <appname>" and use that also together with "app: <appname>" as selector.

I'm afraid that someone will create pod with `app` label and it will be wrongly assigned to a DeploymentConfig